### ISSUE TITLE
Error Error: EventDispatcher is disposed

### ISSUE THREAD

**Issue by prasanna-sasne** [amcharts/amcharts4#4317]
Bug Description

I get sometimes the below error in my console after updating line series data.

Create multiple line series on the chart with decimated data. (Do not show all data at once on the chart)
Now zoom in to the chart and re-add missing data points on series between the selected range. (update chart.data array by adding extra points in that range)

And I get the error "Error Error: EventDispatcher is disposed" 

Environment (if applicable)
amCharts version 4.10.29
Browser and its version: Only detected yet in chrome (because the most used/tested browser)

**Stack**
`core.mjs:9171 ERROR Error: EventDispatcher is disposed
    at EventDispatcher._on (EventDispatcher.js:309:19)
    at EventDispatcher.on (EventDispatcher.js:380:21)
    at Label.setDataItem (Label.js:1345:33)
    at Label.set [as dataItem] (Sprite.js:3308:18)
    at Container.js:1435:17
    at Iterator.js:57:9
    at Iterator.js:10:18
    at Module.each (Iterator.js:56:5)
    at Container.setDataItem (Container.js:1434:13)
    at Tooltip.set [as dataItem] (Sprite.js:3308:18)
<html>
<body>

handleError | @ | core.mjs:9171
-- | -- | --
  | next | @ | core.mjs:26958
  | next | @ | Subscriber.js:91
  | _next | @ | Subscriber.js:60
  | next | @ | Subscriber.js:31
  | (anonymous) | @ | Subject.js:34
  | errorContext | @ | errorContext.js:19
  | next | @ | Subject.js:27
  | emit | @ | core.mjs:22784
  | (anonymous) | @ | core.mjs:26300
  | invoke | @ | zone.js:372
  | run | @ | zone.js:134
  | runOutsideAngular | @ | core.mjs:26173
  | onHandleError | @ | core.mjs:26300
  | handleError | @ | zone.js:376
  | runTask | @ | zone.js:181
  | invokeTask | @ | zone.js:487
  | invokeTask | @ | zone.js:1661
  | globalCallback | @ | zone.js:1692
  | globalZoneAwareCallback | @ | zone.js:1725

</body>
</html>
` 

My use case - 
I'm currently displaying data in a reduced format because using the full dataset freezes the amCharts, as its inability to handle billions of data points (though the data isn't live). To manage this, I've been condensing the data, and during Zoom events, I dynamically adjust the displayed data by adding or removing points. I am also getting the same error when I tried to update the existing chart.data. (I am using Amchart 4 version with angular 15)

However, I'm not entirely satisfied with some solutions given on Git Hub for similar problems[[1](https://github.com/amcharts/amcharts4/issues/3051), [2](https://github.com/amcharts/amcharts4/issues/2967)]. I find it less than ideal to create a new X-axis, assign it to the xAxis property, and update the entire chart for every zoom-in or zoom-out action. I'm looking for a more efficient solution that doesn't require these repeated steps with each Zoom event. Can you please suggest the best way to deal with this use case?

Please note that I don't want to dispose entire chart and re-create it new. I want to update only existing line series values on zoom events.

**Comment by martynasma** [comment]
> update chart.data array by adding extra points in that range

Would you be able to provide more details on how you do it? A code, preferably.

Ideally, it would be best if you could provide a working standalone example which we can test.

**Comment by prasanna-sasne** [comment]
`
// This function initializes the chart data. If there are more than 500 data points, it aggregates them to show 500 data points on the chart initially.
const initializeChartData = () => {
  if (chartData.length > 500) {
    chart.addData(chartData.filter((_, index) => (index % Math.floor(chartData.length / 500) === 0)));
  } else {
    chart.addData(chartData);
  }
}
initializeChartData(); // Initially, aggregated data with up to 500 data points is added to the chart.

// After zooming in, we extract data from the original 'chartData' array based on the specified start and end points of the zoomed-in range.
// We then reduce the data within this range and update the chart with it, keeping the start and end points the same as the original chart.
const zoomEnded = () => {
 const chartStart = this.findMatchingPosition(chartData, timeAxis.minZoomed);
  const chartEnd = this.findMatchingPosition(chartData, timeAxis.maxZoomed);
  const slicedDataPoints = chartData.slice(chartStart, chartEnd + 1);

  if (slicedDataPoints.length > 500) {
    // If there are more than 500 data points in the zoomed range, we decimate them for display.
    const decimatedPoints = slicedDataPoints.filter((_, index) => (index % Math.floor(slicedDataPoints.length / 500) === 0));
    chart.data.splice(1, chart.data.length - 2, ...decimatedPoints);
  } else {
    // Otherwise, we update the chart directly with the sliced data.
    chart.data.splice(1, chart.data.length - 2, ...slicedDataPoints);
  }
  chart.invalidateData(); // This line is causing an error after zooming in 5-6 times.
  timeAxis.zoomToValues(timeAxis.minZoomed, timeAxis.maxZoomed); // Handles zooming of the horizontal scrollbar.
}

// This function handles events related to changes in the start and end of the time axis.
const handleEvent = () => {
  if (startEndTimeout) {
    window.clearTimeout(startEndTimeout);
  }
  startEndTimeout = window.setTimeout(() => zoomEnded(), 100);
}

// Event listeners for changes in the start and end of the time axis trigger the handleEvent function.
timeAxis.events.on("startchanged", handleEvent);
timeAxis.events.on("endchanged", handleEvent);   
`
I am following this [reference link](https://www.amcharts.com/docs/v4/concepts/data/#Manipulating_existing_data_points). 

In summary, this code initializes chart data, updates it after zooming in, and handles events related to changes in the time axis. There's an issue with the chart.invalidateData() line after multiple zooms, which needs further investigation.
    
I will try to reproduce the same error on the code pen, but this is my best-effort code for your reference. Please read the comments given to get more clarity
